src/geo/projection/globe_util.js

```diff
@@ -724,37 +731,117 @@ export class GlobeSharedBuffers {
+                    return x - 2;
+                } else if (x === 0) {
+                    return x;
+                } else {
+                    return x - 1;
                 }
+            })();
+
+            // Skirt factor is introduces as an offset to the .x coordinate, similar to how it's done for mercator grids
+            const skirtOffset = 24575;
+            adjustedX += isSkirt ? skirtOffset : 0;
+
+            return [adjustedX, y];
+        };
+
+        // Add first horizontal strip if present
+        if (embedSkirt) {
+            for (let x = 0; x < xVertices; ++x) {
+                vertices.emplaceBack(...prepareVertex(x, 0, true));
             }
+        }
 
-            const numVertices = (latitudinalLod + 1) * vertexExt;
-            const numPrimitives = latitudinalLod * quadExt * 2;
+        // Add main grid part with vertices strips embedded
+        for (let y = 0; y < yVerticesHighLodNoStrip; ++y) {
+            for (let x = 0; x < xVertices; ++x) {
+                const isSideBorder = (x === 0 || x === xVertices - 1);
 
-            this._gridSegments.push(SegmentVector.simpleSegment(0, primitiveOffset, numVertices, numPrimitives));
-            primitiveOffset += numPrimitives;
+                vertices.emplaceBack(...prepareVertex(x, y, isSideBorder && embedSkirt));
+            }
+        }
+
+        // Add bottom strips for each LOD
+        if (embedSkirt) {
+            for (let lodIdx = 0; lodIdx < latitudinalLods.length; ++lodIdx) {
+                const lastYRowForLod = latitudinalLods[lodIdx];
+                for (let x = 0; x < xVertices; ++x) {
+                    vertices.emplaceBack(...prepareVertex(x, lastYRowForLod, true));
+                }
+            }
         }
 
-        this._gridBuffer = context.createVertexBuffer(gridVertices, posAttributes.members);
-        this._gridIndexBuffer = context.createIndexBuffer(gridIndices, true);
+        // Fill triangles
+        for (let lodIdx = 0; lodIdx < latitudinalLods.length; ++lodIdx) {
+            const indexOffset = indices.length;
+
+            const yVerticesLod = latitudinalLods[lodIdx] + 1 + 2 * (embedSkirt ? 1 : 0);
+
+            for (let y = 0; y < yVerticesLod - 1; y++) {
+                const isLastLine = (y === yVerticesLod - 2);
+                const offsetToNextRow =
+                    (isLastLine && embedSkirt ?
+                        (xVertices * (yVerticesHighLodWithStrip - latitudinalLods.length + lodIdx - y)) :
+                        xVertices);
+
+                for (let x = 0; x < xVertices - 1; x++) {
+                    const idx = y * xVertices + x;
+
+                    indices.emplaceBack(idx + 1, idx, idx + offsetToNextRow);
+                    indices.emplaceBack(idx + offsetToNextRow, idx + offsetToNextRow + 1, idx + 1);
+                }
+            }
+            segments.push(SegmentVector.simpleSegment(0, indexOffset, vertices.length, indices.length - indexOffset));
```

Comment:
Are you including skirt triangles into the same segment as the main grid triangles? We could have two different segments: one with skirts and one without skirts. This way we can skip rendering of skirts if the terrain is not enabled in globe view.

Reply:
If I understood correctly this change will always render skirts?

Reply:
yes, always

Reply:
Otherwise this would mean sending more draw routines to the graphics pipeline. I think it is either

same count of render calls even if that means rendering +6% extra degenerate triangles

or 

double the amount of render calls

Reply:
would it be possible to construct index buffer so that skirt triangles are immediately after terrain triangles? This wouldn't increase draw count as the only difference would be the number of rendered indices?

Reply:
currently, always, but yeah two segments (one with skirt and one without) can be used depending on terrain exaggeration. I'll do that modification

Reply:
Done